src/steamship/agents/functional/output_parser.py

```diff
@@ -1,13 +1,17 @@
 import json
 import re
+import string
 from typing import Dict, List, Optional
 
 from steamship import Block, MimeTypes, Steamship
 from steamship.agents.schema import Action, AgentContext, FinishAction, OutputParser, Tool
 from steamship.data.tags.tag_constants import RoleTag
 
 
-# TODO(dougreid): extract shared bits from this and the ReACT output parser into a utility?
+def is_punctuation(text: str):
+    return text in string.punctuation
```

Comment:
Do you want this to return true if ALL the text is punctuation, or only if it is a single character of punctuation?  The current implementation does the latter:
```
>>> s = ".."
>>> import string
>>> s in string.punctuation
False
```

Reply:
good point. updated. still doesn't filter strings like `....   ....   ...`, but that is so degenerate a case that I'm not sure we need to worry about it.